CMakeLists.txt

```diff
@@ -484,49 +484,47 @@ if(AVIF_CODEC_LIBRARIES MATCHES vmaf)
 endif()
 
 option(AVIF_BUILD_APPS "Build avif apps." OFF)
-if(AVIF_BUILD_APPS)
+option(AVIF_BUILD_TESTS "Build avif tests." OFF)
+
+if(AVIF_BUILD_APPS OR AVIF_BUILD_TESTS)
+    # Not all targets depending on avifappsshared, such as tests, need png and jpeg.
+    # However it is simpler this way and likely that a test environment provides these.
```

Comment:
Nit: I think we can omit this comment. The new library must depend on png and jpeg because we decided to put avifjpeg.c and avifpng.c in the new library. Also, if new tests use png or jpeg in the future, this comment will become stale.